### PR TITLE
RESTWS-736: ObsResource.setValue should support map values for coded values

### PR DESCRIPTION
**Pull request by wluyima**

### CODE REVIEW

**Comment by dkayiwa** [comment]
@wluyima did you see the merge conflicts on this?

**Comment by wluyima** [comment]
Resolved

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java [hunk 1]
```diff
@@ -392,6 +382,16 @@ public static void setValue(Obs obs, Object value) throws ParseException, Conver
...
+				//We want clients to be able to fetch a coded value in one rest call
+				//and set the returned payload as the obs value
+				if (value instanceof Map) {
+					Object uuid = ((Map) value).get(RestConstants.PROPERTY_UUID);
+					if (uuid != null) {
+						value = uuid.toString();
+					}
+				}
+				
 				Concept valueCoded = (Concept) ConversionUtil.convert(value, Concept.class);
```
Comment (dkayiwa) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java]: The ticket description seems to discourage using the above. It instead recommends ConversionUtil.getConverter(Concept.class).getByUniqueId(uuid). Did i just misunderstand it?

Comment (wluyima) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java]: After taking a deeper look, I realized the issue was a little different and I was supposed to update the ticket description which I have done to remove the need to make that change.

**Pull request merged**